modules/giapi/src/main/scala/giapi/client/package.scala

```diff
@@ -157,42 +157,67 @@ package client {
         } yield i
       }
 
+    // taken from cats-effect 1.x
+    private def toIOFromRunAsync[F[_], A](f: F[A])(implicit F: Effect[F]): IO[A] =
+      IO.async { cb =>
+        F.runAsync(f)(r => IO(cb(r))).unsafeRunSync()
+      }
+
+    // Implementations of timeout suggested from cats-effect documentation
+    private def timeoutTo[F[_]: Effect, A](fa: F[A], after: FiniteDuration, fallback: F[A])
+                    (implicit timer: Timer[IO]): F[A] = {
+
+      // Race the task against a sleep timer
+      LiftIO[F].liftIO(IO.race(toIOFromRunAsync(fa), timer.sleep(after)).flatMap {
+        case Left(a) => IO.pure(a)
+        case Right(_) => toIOFromRunAsync(fallback)
+      })
+    }
+
+    private def timeout[F[_]: Effect, A](fa: F[A], after: FiniteDuration, error: Exception)
+                  (implicit timer: Timer[IO]): F[A] = {
+
+      timeoutTo(fa, after, Effect[F].raiseError(error))
+    }
+
     /**
       * Interpreter on F
       *
       * @param url Url to connect to
       * @tparam F Effect type
       */
     // scalastyle:off
-    def giapiConnection[F[_]: Effect](url: String, commandsTimeout: Duration): GiapiConnection[F] =
+    def giapiConnection[F[_]: Effect](
+        url: String,
+        ec: ExecutionContext): GiapiConnection[F] =
       new GiapiConnection[F] {
         private def giapi(c: ActiveMQJmsProvider,
                           sg: StatusGetter,
                           cc: CommandSenderClient,
                           ss: StatusStreamer) =
           new Giapi[F] {
-            @SuppressWarnings(Array("org.wartremover.warts.Throw"))
+            private val commandsAckTimeout                  = 2000.milliseconds
+            implicit val executionContext: ExecutionContext = ec
+
             override def get[A: ItemGetter](statusItem: String): F[A] =
-              Sync[F].delay {
-                val item = sg.getStatusItem[A](statusItem)
-                // Note item.getValue can throw if e.g. the item is unknown, let's avoid NPEs though
-                Option(item) match {
-                  case Some(i) => i.getValue
-                  case None    => throw GiapiException(s"Status item $statusItem not found")
-                }
+              getO[A](statusItem).flatMap {
+                case Some(a) => a.pure[F]
+                case None =>
+                  Sync[F].raiseError(
+                    new GiapiException(s"Status item $statusItem not found"))
```

Comment:
FYI you can say `OptionT(getO[A](statusItem)).getOrElseF[A](Sync[F].raiseError(...)).value` if that seems better. What you have here is fine though.